conan/tools/system/package_manager.py

```diff
@@ -0,0 +1,175 @@
+class SystemPackageManagerTool(object):
+    install_methods = ["install", "update"]
+    mode_check = "check"
+    tool_name = None
+    install_command = ""
+    update_command = ""
+    check_command = ""
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._active_tool = self._conanfile.conf["tools.system.package_manager:tool"]
+        self._sudo = self._conanfile.conf["tools.system.package_manager:sudo"]
+        self._sudo_askpass = self._conanfile.conf["tools.system.package_manager:sudo_askpass"]
+        self._mode = self._conanfile.conf["tools.system.package_manager:mode"] or self.mode_check
+        self._arch = self._conanfile.settings.get_safe("arch")
+        self._arch_names = {}
+        self._arch_separator = ""
+
+    def get_package_name(self, package):
+        # TODO: should we only add the arch if cross-building?
+        if self._arch in self._arch_names:
+            return "{}{}{}".format(package, self._arch_separator,
+                                   self._arch_names.get(self._arch))
+        return package
+
+    @property
+    def sudo_str(self):
+        sudo = "sudo " if self._sudo else ""
+        askpass = "-A " if self._sudo and self._sudo_askpass else ""
+        return "{}{}".format(sudo, askpass)
+
+    def method_decorator(wrapped):
```

Comment:
I think at some point, this should fail. If you use one of this, the mode is not "install" but only "check", and the packages are not installed in the system, it should raise an exception, not just warn?

We might want to re-design this a bit more decoupled, having a method that only collects the needs (like ``requirements()`` do for Conan dependencies. But the execution only happens later. This way we can collect, list, report on system requirements easily (besides installing them). This would be a different feature, probably out of the scope of this, but just an idea, this has been requested in the past by users (wanting to know, for a full dependency graph, what "system-requirements" are needed, etc, before installing them)

Reply:
I changed it to raise an exception now, I also added the option of doing a check of missing packages before installing...